src/codegen/sys/build.rs

```diff
@@ -103,28 +103,11 @@ fn find() -> Result<(), Error> {
         return Ok(())
     }
 
-    let target = env::var("TARGET").expect("TARGET environment variable doesn't exist");
-    let hardcode_shared_libs = target.contains("windows");
-
     let mut config = Config::new();
     config.atleast_version(version);
     config.print_system_libs(false);
-    if hardcode_shared_libs {
-        config.cargo_metadata(false);
-    }
     match config.probe(package_name) {
-        Ok(library) => {
-            if hardcode_shared_libs {
-                for lib_ in shared_libs.iter() {
-                    println!("cargo:rustc-link-lib=dylib={}", lib_);
-                }
-                for path in library.link_paths.iter() {
-                    println!("cargo:rustc-link-search=native={}",
-                             path.to_str().expect("library path doesn't exist"));
-                }
```

Comment:
For me only this for can be removed even after upgrading pkg-config to 0.3.14, but not sure that it works globally as I have msys64/bin with gtk's dlls in path.
I tested on examples\cairo_threads.rs, do this change works fine for cairo-sys.
But on gtk it not works:
removing dylib caused many
```
          D:\eap\rust\0\target\debug\deps\libgtk-d3ee1e1d9d9104f6.rlib(gtk-d3ee1e1d9d9104f6.1consvywiv34sg52.rcgu.o): In function `_$LT$gtk..border..MemoryManager$u20$as$u20$glib..boxed..BoxedMemoryManager$LT$gtk_sys..GtkBorder$GT$$GT$::free::h324bcaf1ec9c7496':
          D:/eap/rust/0/gtk\src/border.rs:20: undefined reference to `gtk_border_free'
```
removing `config.cargo_metadata(false);` caused
```
  = note: ld: cannot find -limm32
          ld: cannot find -ldwmapi
          ld: cannot find -lcfgmgr32
```

Reply:
Any idea why? Can you provide the corresponding (for each issue) cargo output (with `-vv`) of a clean build?

Reply:
IMHO at minimum no working gtk static lib on windows, about others I don't have any idea.
Output here: https://gist.github.com/EPashkin/a07f84a48f1b791e382e22716e2da144

Reply:
Why would it try static libs to begin with?

The GTK undefined symbols happen because there's no `-lgtk-3` in the linker commandline. The gtk-sys build.rs does *not* output this (or anything) for whatever reason. Can you check why? The other -sys crates output the correct things.

The missing Windows libraries, can you check where they are on your system and why those directories (the ones with the correspnding .lib files) are not in the linker search path?